### PR TITLE
Handle exception when creating a reference immediately after creating fork

### LINKED ISSUE #130: WingetCreate update command failing during Pull Request submission due to issue with permission
# Brief description of your issue

I use WingetCreate update command in a [GitHub Actions workflow](https://github.com/nushell/nushell/actions/workflows/winget-submission.yml)  that aims at generating the manifest and submitting the PR to the winget packages repository. The manifest generation works fine but the Pull Request submission fails with a permission error although the PAT was generated with the public_repo permission: `Octokit.ForbiddenException: Must have admin rights to Repository.`

# Steps to reproduce

- Create a PAT with the public_repo permission, set it in the secrets of a GitHub repository ([Nushell](https://github.com/nushell/nushell) in my case with a secret named `NUSHELL_PAT`)
- Use that secret with the wingetcreate update command in a github actions workflow 
My workflow is the following (triggered by a release):
```yaml
name: Submit Nushell package to Windows Package Manager Community Repository 

on:
  release:
    types: [published]

jobs:

  winget:
    name: Publish winget package
    runs-on: windows-latest
    steps:
      - name: Submit package to Windows Package Manager Community Repository
        run: |
          iwr https://aka.ms/wingetcreate/latest -OutFile wingetcreate.exe
          $github = Get-Content '${{ github.event_path }}' | ConvertFrom-Json
          $installerUrl = $github.release.assets | Where-Object -Property name -match 'windows.msi' | Select -ExpandProperty browser_download_url -First 1
          .\wingetcreate.exe update Nushell.Nushell -s -v $github.release.tag_name -u $installerUrl -t ${{ secrets.NUSHELL_PAT }}
```

# Expected behavior

Update command should work and the PR should be created like in the following screenshot
image
(the same github action script worked on my fork of the main repository)

# Actual behavior

Pull Request submission is failing with the following error:
image

This raises the following questions:
- Is the public_repo permission not enough to make wingetcreate work ? => It would be weird as I succeeded making it work with this permission in another fork.
- Should the creator of the PAT be the owner of the original repo (Nushell in my case) ? I don't see why as the submission is mainly about forking winget-pkgs and submitting a PR if I understood correctly.

It seems to me that what is failing is the push of the autogenerated branch to the winget-pkgs fork. But I don't know what is the problem exactly.

# Environment

Latest version of winget create is used (latest version is downloaded each time the workflow is run): currently 0.2.0.29

### PR DESCRIPTION
**Pull request by ryfu-msft**
Fixes #130 

Description of Bug:
A weird behavior occurs when trying to create a reference to a forked repo immediately after creating the forked repo. This issue does not repro if the forked winget-pkgs repo had already existed previously. 

The error in the linked issue #130 is actually caused by the issue stated above but is inaccurately reported because the user did not provide a token with "delete_repo" permission, which is needed to clean up the forked repo if there is a failure that occurs during PR submission.

Changes:
- Adds a wait and retry policy when creating a reference to handle this exception.
- Catch the forbidden exception that is thrown if a user does not provide a token with the "delete_repo" access.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-create/pull/143)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,6 +52,17 @@ Then simply add a new powershell task to download the exe, and run it to update
 
 The CLI also supports creating or updating manifests with multiple installer URLs. You can either create new manifests with multiple installer nodes using the [New Command](doc/new.md) or update existing manifests with multiple installer URLs using the [Update Command](doc/update.md). 
 
+## Using your GitHub Personal Access Token
+
+When [creating your own GitHub Personal Access Token (PAT)](https://docs.github.com/en/github/authenticating-to-github/keeping-your-account-and-data-secure/creating-a-personal-access-token) to be used with WingetCreate, make sure you verify the following steps are completed:
+
+- [Authorizing your personal access token for use with SAML single sign-on](https://docs.github.com/en/github/authenticating-to-github/authenticating-with-saml-single-sign-on/authorizing-a-personal-access-token-for-use-with-saml-single-sign-on)
```
Comment (palenshus) [inline README.md:59]: Are you sure everyone needs to do this, or just us MSers?

File: src/WingetCreateCore/Common/GitHub.cs [hunk 2]
```diff
@@ -307,8 +308,11 @@ private async Task<PullRequest> SubmitPRAsync(string packageId, string version,
             Reference newBranch = null;
             try
             {
-                // Create new branch synced to upstream master
-                await this.github.Git.Reference.Create(repo.Id, new NewReference($"refs/{newBranchNameHeads}", upstreamMasterSha));
+                var retryPolicy = Policy.Handle<ApiException>().WaitAndRetryAsync(3, i => TimeSpan.FromSeconds(2));
```
Comment (palenshus) [inline src/WingetCreateCore/Common/GitHub.cs:311]: I'd change this slightly to TimeSpan.FromSeconds(i)

That gives us a bit of a linear backoff with fastFirst behavior ([https://github.com/Polly-Contrib/Polly.Contrib.WaitAndRetry/#retry-first-failure-fast](https://github.com/Polly-Contrib/Polly.Contrib.WaitAndRetry/#retry-first-failure-fast))

**Review by palenshus — CHANGES_REQUESTED** [review]
:clock1:

File: README.md (hunk 1 above)
Comment (ryfu-msft) [reply README.md]: Oops, you're right. Just us MSers. Deleted.

File: src/WingetCreateCore/Common/GitHub.cs (hunk 2 above)
Comment (ryfu-msft) [reply src/WingetCreateCore/Common/GitHub.cs]: Changed to suggested

**Review by palenshus — APPROVED** [review]
:shipit:

**Pull request merged**